src/test/java/apicurito/tests/utils/slenide/CommonUtils.java

```diff
@@ -81,11 +90,56 @@ public static void setValueInLabel(String value, SelenideElement section, boolea
     }
 
     public static void setDropDownValue(String buttonId, String value, SelenideElement section) {
-        log.info("Setting value {} in dropdown {} in section{}", value, buttonId, section.getValue());
+        log.info("Setting value {} in dropdown {} in section {}", value, buttonId, section.getAttribute("class"));
         section.$(buttonId).click();
         section.$$("li").filter(text(value)).first().click();
     }
 
+    /**
+     * It works for creating: Query parameters (both), Header parameters (both), Reqest form data, Data type property
+     */
+    public static void fillEntityEditorForm(DataTable table) {
+        SelenideElement newDataFormPage = MainPageUtils.getMainPageRoot().$("#entity-editor-form");
+
+        for (List<String> dataRow : table.cells()) {
+            newDataFormPage.$("#qp_name").setValue(dataRow.get(0));
+
+            if (!dataRow.get(1).isEmpty()) {
+                newDataFormPage.$("ace-editor textarea").sendKeys(dataRow.get(1));
+            }
+
+            if (!dataRow.get(2).isEmpty()) {
+                CommonUtils.setDropDownValue(CommonUtils.DropdownButtons.PARAM_REQUIRED.getButtonId(), Boolean.valueOf(dataRow.get(2)) ? "Required" : "Not Required", newDataFormPage);
```

Comment:
Why don't you put the value `Required` or `Not Required` straight into gherkin step string parameter instead of using boolean? Then if you need another logic you can use string comparison... Please change it on all places, you use it often.

Reply:
updated